practical_07/project_management.py

```diff
@@ -0,0 +1,207 @@
+"""
+
+Estimated completion time: 3h
+Actual completion time:
+"""
+
+from project import Project
+from datetime import datetime
+from operator import attrgetter
+
+NAME_INDEX = 0
+START_DATE_INDEX = 1
+PRIORITY_INDEX = 2
+COST_ESTIMATE_INDEX = 3
+COMPLETION_PERCENTAGE_INDEX = 4
+
+MENU = """
+- (L)oad projects  
+- (S)ave projects  
+- (D)isplay projects  
+- (F)ilter projects by date
+- (A)dd new project  
+- (U)pdate project
+- (Q)uit
+
+>>> """
+
+
+def main():
+    menu_choice = input(MENU).upper()
+    while menu_choice != "Q":
+        if menu_choice == "L":
+            projects = load_projects()
+        elif menu_choice == "S":
+            try:
+                save_projects(projects)
+            except UnboundLocalError:
+                print("Projects must first be loaded")
```

Comment:
I see you handled what I had previously mentioned here. There's a lot of try-except suites here.